reccmp/isledecomp/compare/asm/instgen.py

```diff
@@ -24,9 +23,20 @@ class SectionType(Enum):
     ADDR_TAB = auto()
 
 
-class FuncSection(NamedTuple):
-    type: SectionType
-    contents: List[Union[DisasmLiteInst, Tuple[str, int]]]
+class CodeSection(NamedTuple):
+    type: Literal[SectionType.CODE]
+    contents: List[DisasmLiteInst]
+
+
+TabSectionType = Literal[SectionType.DATA_TAB] | Literal[SectionType.ADDR_TAB]
+
+
+class TabSection(NamedTuple):
+    type: TabSectionType
+    contents: List[Tuple[int, int]]
+
+
+FuncSection = CodeSection | TabSection
```

Comment:
This is effectively what the code did before. The difference is that you can check for `type` now and then deduce the type of `contents`.